jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -261,11 +261,11 @@ def segwit_signature_form(txobj, i, script, amount, hashcode=SIGHASH_ALL,
 def signature_form(tx, i, script, hashcode=SIGHASH_ALL):
     if not isinstance(hashcode, int):
         hashcode = struct.unpack(b'B', hashcode)[0]
-    if isinstance(tx, basestring) and not isinstance(tx, bytes) and re.match('^[0-9a-fA-F]*$', tx):
+    if isinstance(tx, basestring) and not isinstance(tx, bytes):
```

Comment:
Here we should change the logic imo: `if not isinstance(tx, dict): tx = deserialize(tx)` - and remove all these other checks. Logic: `deserialize` itself checks the encoding, so better not to repeat that check. The only options are to call this (signature form) with a dict already, or to have it deserialized on the fly.

As a point of interest, this is currently only used in one test: `test_coinjoin.py` in jmclient, but ofc we need/want it to work.

Reply:
is the non-dict input for this always hex encoded?

Reply:
If tx comes in as bytes, or as a string, `deserialize` will check that.

If it comes in as a string, but not a hex-encoded string, then that's an error condition (as we were discussing on IRC, that's up to callers to not do).

Reply:
> If tx comes in as bytes, or as a string, deserialize will check that.

The issue I think is that the output format for `deserialize` is dependent upon whether or not it comes in as a string or bytes, hence why `signature_form` is ensuring that a hex string is passed to `deserialize`(since `signature_form` is only designed to handle hex).

Reply:
Yes, thank you.
We really need to nuke all this mutation :) Every function here should just fix what types it takes in ... but, for another time.